tools/cbmc/proofs/ARP/ARPGenerateRequestPacket/ARPGenerateRequestPacket_harness.c

```diff
@@ -0,0 +1,29 @@
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "queue.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_IP_Private.h"
+#include "FreeRTOS_ARP.h"
+
+void harness()
+{
+	ARPPacket_t xArpPacket;
+	NetworkBufferDescriptor_t xNetworkBuffer;
+	xNetworkBuffer.pucEthernetBuffer = &xArpPacket;
+	vARPGenerateRequestPacket( &xNetworkBuffer );
+
+	/*
+	 * The assumption made here is that the buffer pointed by pucEthernerBuffer
+	 * is at least allocated to sizeof(ARPPacket_t) size.
+	 * This is not checked inside vARPGenerateRequestPacket.
+	 */
+	uint8_t ucBUFFER_SIZE;
+	__CPROVER_assume( ucBUFFER_SIZE > sizeof(ARPPacket_t) && ucBUFFER_SIZE < 100 );
+	void *xBuffer = malloc(ucBUFFER_SIZE);
+
+	NetworkBufferDescriptor_t xNetworkBuffer2;
+	xNetworkBuffer2.pucEthernetBuffer = xBuffer;
+	vARPGenerateRequestPacket( &xNetworkBuffer2 );
```

Comment:
This test catches everything the first test would catch, and I consider this a superior implementation.